deb/debian/stackdriver-agent.init

```diff
@@ -132,64 +115,48 @@ d_start() {
     fi
 
     if ! check_config; then
-        log_failure_msg "Not starting, configuration/credentials error."
+        log_failure_msg "not starting, configuration/credentials error."
         return 3
     fi
 
-    if test "$USE_COLLECTDMON" == 1; then
-        LD_LIBRARY_PATH="$LD_LIBRARY_PATH:/opt/stackdriver/collectd/lib64:/opt/stackdriver/collectd/lib" start-stop-daemon --start --quiet --oknodo --pidfile "$_PIDFILE" \
-            --exec $COLLECTDMON_DAEMON -- -P "$_PIDFILE" -- -C "$CONFIG" \
-            || return 2
-    else
-        LD_LIBRARY_PATH="$LD_LIBRARY_PATH:/opt/stackdriver/collectd/lib64:/opt/stackdriver/collectd/lib" start-stop-daemon --start --quiet --oknodo --pidfile "$_PIDFILE" \
-            --exec "$DAEMON" -- -C "$CONFIG" -P "$_PIDFILE" \
-            || return 2
-    fi
+    LD_LIBRARY_PATH="$LD_LIBRARY_PATH:/opt/stackdriver/collectd/lib64:/opt/stackdriver/collectd/lib" \
+        start_daemon -p "$_PIDFILE" "$DAEMON" -C "$CONFIG" -P "$_PIDFILE" \
+        || return 2
     return 0
 }
 
-still_running_warning="
-WARNING: $NAME might still be running.
-In large setups it might take some time to write all pending data to
-the disk. You can adjust the waiting time in /etc/default/$NAME."
-
 # return:
 #   0 if the daemon has been stopped
-#   1 if the daemon was already stopped
-#   2 if daemon could not be stopped
-d_stop() {
-    PID=$( cat "$_PIDFILE" 2> /dev/null ) || true
-
-    start-stop-daemon --stop --quiet --oknodo --pidfile "$_PIDFILE"
-    rc="$?"
-
-    if test "$rc" -eq 2; then
-        return 2
-    fi
+#   1 if the daemon could not be stopped
+#   3 if the daemon was already stopped
+stop() {
+    killproc -p "$_PIDFILE" "$DAEMON"
+}
 
-    sleep 1
-    if test -n "$PID" && kill -0 $PID 2> /dev/null; then
-        i=0
-        while kill -0 $PID 2> /dev/null; do
-            i=$(( $i + 2 ))
-            echo -n " ."
-
-            if test $i -gt $MAXWAIT; then
-                log_progress_msg "$still_running_warning"
-                return 2
-            fi
-
-            sleep 2
-        done
-        return "$rc"
+# return:
+#   0 if the daemon has been restarted
+#   1 if the daemon could not be stopped
+#   2 if the daemon could not be started
+#   3 if the daemon was not supposed to be restarted
+restart() {
+    if ! check_config; then
+        log_failure_msg "not restarting, configuration/credentials error"
+        return 3
     fi
-    return "$rc"
+    stop
+    rc="$?"
+    case "$rc" in
+        0|3) ;;
+        *) return "$rc" ;;
+    esac
+    start
 }
 
+# See how we were called.
 case "$1" in
     start)
         log_daemon_msg "Starting $DESC" "$NAME"
-        d_start
+        start
         case "$?" in
             0|1) log_end_msg 0 ;;
```

Comment:
Looks like we will no longer return `1` in any cases?

Reply:
I don't think we ever did (the comment actually documented the return values from `start-stop-daemon`, which our script overrides), but leaving it here is harmless for symmetry with other cases.

Reply:
Ah i see. Sounds good.